files.py

```diff
@@ -1,21 +1,26 @@
 import lp_colors, scripts
 from time import sleep
 import os, json, platform, subprocess
+from constants import *
 
 LAYOUT_DIR = "user_layouts"
 SCRIPT_DIR = "user_scripts"
+SUBROUTINE_DIR = "user_subroutines"
 
-FILE_VERSION = "0.1.1"
+FILE_VERSION = "0.1.1"         # file is unchanged if no subroutines are saved
+FILE_VERSION_SUBS = "0.2"      # file version if subroutines are saved
 
 LAYOUT_EXT = ".lpl"
 SCRIPT_EXT = ".lps"
+SUBROUTINE_EXT = ".lpc"

```

Comment:
Why can't this be saved in the same layout file?

Reply:
It is.  That's the name of files which contain subroutines.  You don't always have to load them.  Also if you want to have the same subroutines in 2 or more layouts, you don't want to be maintaining 2 copies of the source.